MapboxMobileEvents/MMEEvent.m

```diff
@@ -160,6 +160,17 @@ + (instancetype)carplayEventWithName:(NSString *)name attributes:(NSDictionary *
     return carplayEvent;
 }
 
++ (instancetype)eventWithDateString:(NSString *)dateString name:(NSString *)name attributes:(NSDictionary *)attributes {
```

Comment:
Just a couple of questions here (which I think we've talked about before briefly):

1. Why is this a class method rather than an `MMEEvent` initializer? Is this for future support for different `MMEEvent` subclasses? So does this mean `-[MMEEvent init]` is essentially "private"? If so, I wonder if it's worth having a specific initializer (in addition) and marking the default `init` as `NS_UNAVAILABLE`. Thoughts? 
2. Should this method take an `NSDate` rather than a date string? Perhaps a date transformer could be supplied (somewhere).

Reply:
This PR is somewhat of a stopgap measure until we can completely remove the whitelisting that is currently in place. The [doc](https://paper.dropbox.com/doc/iOS-Removing-Whitelist-from-event-queuing--AU~Rjc1CqFqIRDaJTQ7GKL4wAg-Uqob9uEUWBF0t67RUuGTD) describes some of those plans and how we might accomplish them. 

1. I'm not sure how the decision came about to make these class methods or what the future plans were for it. I don't see any harm in a user of the library using an initializer vs any of these class methods to construct an event however. Is there something I'm missing here?
2. I agree that this could be provided in another way and potentially better way (especially once we implement date/time correction {insert Doctor Who meme}) but I think that should be done a little further down the road. For now the date string is supplied by the `eventsManager` via [`MMENSDateWrapper`](https://github.com/mapbox/mapbox-events-ios/blob/master/MapboxMobileEvents/MMENSDateWrapper.m#L27-L29) and all that needs to be called by the client/host is [`enqueueEventWithName:attributes:`](https://github.com/mapbox/mapbox-events-ios/blob/master/MapboxMobileEvents/MMEEventsManager.m#L353-L394).

Reply:
> Is there something I'm missing here?

No - more curious about future plans 😄 

> [...] but I think that should be done a little further down the road

Thanks for clarifying 👍